// ng2-charts/components/charts/charts.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var import0 = require('@angular/core/src/linker/ng_module_factory');
var import1 = require('ng2-charts/components/charts/charts');
var import4 = require('@angular/core/src/linker/view');
var import5 = require('@angular/core/src/linker/element');
var import7 = require('@angular/core/src/linker/view_type');
var import8 = require('@angular/core/src/change_detection/change_detection');
var import9 = require('@angular/core/src/linker/element_ref');
var import10 = require('@angular/core/src/metadata/view');
var import11 = require('@angular/core/src/linker/component_factory');
var ChartsModuleInjector = (function (_super) {
    __extends(ChartsModuleInjector, _super);
    function ChartsModuleInjector(parent) {
        _super.call(this, parent, [], []);
    }
    ChartsModuleInjector.prototype.createInternal = function () {
        this._ChartsModule_0 = new import1.ChartsModule();
        return this._ChartsModule_0;
    };
    ChartsModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import1.ChartsModule)) {
            return this._ChartsModule_0;
        }
        return notFoundResult;
    };
    ChartsModuleInjector.prototype.destroyInternal = function () {
    };
    return ChartsModuleInjector;
}(import0.NgModuleInjector));
exports.ChartsModuleNgFactory = new import0.NgModuleFactory(ChartsModuleInjector, import1.ChartsModule);
var renderType_BaseChartComponent_Host = null;
var _View_BaseChartComponent_Host0 = (function (_super) {
    __extends(_View_BaseChartComponent_Host0, _super);
    function _View_BaseChartComponent_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_BaseChartComponent_Host0, renderType_BaseChartComponent_Host, import7.ViewType.HOST, viewUtils, parentInjector, declarationEl, import8.ChangeDetectorStatus.CheckAlways);
    }
    _View_BaseChartComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('base-chart', rootSelector, null);
        this._appEl_0 = new import5.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_BaseChartComponent0(this.viewUtils, this.injector(0), this._appEl_0);
        this._BaseChartComponent_0_4 = new import1.BaseChartComponent(new import9.ElementRef(this._el_0));
        this._appEl_0.initComponent(this._BaseChartComponent_0_4, [], compView_0);
        compView_0.create(this._BaseChartComponent_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_BaseChartComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import1.BaseChartComponent) && (0 === requestNodeIndex))) {
            return this._BaseChartComponent_0_4;
        }
        return notFoundResult;
    };
    _View_BaseChartComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (((this.numberOfChecks === 0) && !throwOnChange)) {
            this._BaseChartComponent_0_4.ngOnInit();
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_BaseChartComponent_Host0.prototype.destroyInternal = function () {
        this._BaseChartComponent_0_4.ngOnDestroy();
    };
    return _View_BaseChartComponent_Host0;
}(import4.AppView));
function viewFactory_BaseChartComponent_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_BaseChartComponent_Host === null)) {
        (renderType_BaseChartComponent_Host = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, [], {}));
    }
    return new _View_BaseChartComponent_Host0(viewUtils, parentInjector, declarationEl);
}
exports.BaseChartComponentNgFactory = new import11.ComponentFactory('base-chart', viewFactory_BaseChartComponent_Host0, import1.BaseChartComponent);
var styles_BaseChartComponent = ['[_nghost-%COMP%] { display: block; }'];
var renderType_BaseChartComponent = null;
var _View_BaseChartComponent0 = (function (_super) {
    __extends(_View_BaseChartComponent0, _super);
    function _View_BaseChartComponent0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_BaseChartComponent0, renderType_BaseChartComponent, import7.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import8.ChangeDetectorStatus.CheckAlways);
    }
    _View_BaseChartComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'canvas', null);
        this.renderer.setElementAttribute(this._el_0, 'style', 'width: 100%; height: 100%;');
        this.init([], [this._el_0], [], []);
        return null;
    };
    return _View_BaseChartComponent0;
}(import4.AppView));
function viewFactory_BaseChartComponent0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_BaseChartComponent === null)) {
        (renderType_BaseChartComponent = viewUtils.createRenderComponentType('/Users/paulo/Documents/workspaces/go/src/github.com/prsolucoes/goci/web-app/node_modules/ng2-charts/components/charts/charts.d.ts class BaseChartComponent - inline template', 0, import10.ViewEncapsulation.Emulated, styles_BaseChartComponent, {}));
    }
    return new _View_BaseChartComponent0(viewUtils, parentInjector, declarationEl);
}
exports.viewFactory_BaseChartComponent0 = viewFactory_BaseChartComponent0;
