// @angular/forms/src/directives/ng_model_group.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/forms/src/directives/ng_model_group');
var import1 = require('@angular/core/src/change_detection/change_detection');
var import2 = require('@angular/core/src/linker/view_utils');
var Wrapper_NgModelGroup = (function () {
    function Wrapper_NgModelGroup(p0, p1, p2) {
        this.changed = false;
        this.context = new import0.NgModelGroup(p0, p1, p2);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_NgModelGroup.prototype.check_name = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.name = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgModelGroup.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_NgModelGroup.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgModelGroup;
}());
exports.Wrapper_NgModelGroup = Wrapper_NgModelGroup;
