// @angular/forms/src/directives/radio_control_value_accessor.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/forms/src/directives/radio_control_value_accessor');
var import1 = require('@angular/core/src/change_detection/change_detection');
var import2 = require('@angular/core/src/linker/view_utils');
var Wrapper_RadioControlValueAccessor = (function () {
    function Wrapper_RadioControlValueAccessor(p0, p1, p2, p3) {
        this.changed = false;
        this.context = new import0.RadioControlValueAccessor(p0, p1, p2, p3);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
    }
    Wrapper_RadioControlValueAccessor.prototype.check_name = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.name = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_RadioControlValueAccessor.prototype.check_formControlName = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this.changed = true;
            this.context.formControlName = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_RadioControlValueAccessor.prototype.check_value = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this.changed = true;
            this.context.value = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_RadioControlValueAccessor.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        if (!throwOnChange) {
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    Wrapper_RadioControlValueAccessor.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_RadioControlValueAccessor;
}());
exports.Wrapper_RadioControlValueAccessor = Wrapper_RadioControlValueAccessor;
