// @angular/forms/src/directives/reactive_directives/form_group_name.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/forms/src/directives/reactive_directives/form_group_name');
var import1 = require('@angular/core/src/change_detection/change_detection');
var import2 = require('@angular/core/src/linker/view_utils');
var Wrapper_FormGroupName = (function () {
    function Wrapper_FormGroupName(p0, p1, p2) {
        this.changed = false;
        this.context = new import0.FormGroupName(p0, p1, p2);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_FormGroupName.prototype.check_name = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.name = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_FormGroupName.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_FormGroupName.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_FormGroupName;
}());
exports.Wrapper_FormGroupName = Wrapper_FormGroupName;
var Wrapper_FormArrayName = (function () {
    function Wrapper_FormArrayName(p0, p1, p2) {
        this.changed = false;
        this.context = new import0.FormArrayName(p0, p1, p2);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_FormArrayName.prototype.check_name = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.name = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_FormArrayName.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        if (!throwOnChange) {
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    Wrapper_FormArrayName.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_FormArrayName;
}());
exports.Wrapper_FormArrayName = Wrapper_FormArrayName;
