// @angular/common/src/directives/ng_class.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/common/src/directives/ng_class');
var import1 = require('@angular/core/src/change_detection/change_detection');
var import2 = require('@angular/core/src/linker/view_utils');
var Wrapper_NgClass = (function () {
    function Wrapper_NgClass(p0, p1, p2, p3) {
        this.changed = false;
        this.context = new import0.NgClass(p0, p1, p2, p3);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
    }
    Wrapper_NgClass.prototype.check_klass = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.klass = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgClass.prototype.check_ngClass = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this.changed = true;
            this.context.ngClass = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_NgClass.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        if (!throwOnChange) {
            this.context.ngDoCheck();
        }
        return changed;
    };
    Wrapper_NgClass.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgClass;
}());
exports.Wrapper_NgClass = Wrapper_NgClass;
