// @angular/common/src/directives/ng_plural.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/common/src/directives/ng_plural');
var import1 = require('@angular/core/src/change_detection/change_detection');
var import2 = require('@angular/core/src/linker/view_utils');
var Wrapper_NgPlural = (function () {
    function Wrapper_NgPlural(p0) {
        this.changed = false;
        this.context = new import0.NgPlural(p0);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_NgPlural.prototype.check_ngPlural = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.ngPlural = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgPlural.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_NgPlural.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgPlural;
}());
exports.Wrapper_NgPlural = Wrapper_NgPlural;
var Wrapper_NgPluralCase = (function () {
    function Wrapper_NgPluralCase(p0, p1, p2, p3) {
        this.changed = false;
        this.context = new import0.NgPluralCase(p0, p1, p2, p3);
    }
    Wrapper_NgPluralCase.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_NgPluralCase.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgPluralCase;
}());
exports.Wrapper_NgPluralCase = Wrapper_NgPluralCase;
