// @angular/common/src/directives/ng_switch.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/common/src/directives/ng_switch');
var import1 = require('@angular/core/src/change_detection/change_detection');
var import2 = require('@angular/core/src/linker/view_utils');
var Wrapper_NgSwitch = (function () {
    function Wrapper_NgSwitch() {
        this.changed = false;
        this.context = new import0.NgSwitch();
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_NgSwitch.prototype.check_ngSwitch = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.ngSwitch = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgSwitch.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_NgSwitch.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgSwitch;
}());
exports.Wrapper_NgSwitch = Wrapper_NgSwitch;
var Wrapper_NgSwitchCase = (function () {
    function Wrapper_NgSwitchCase(p0, p1, p2) {
        this.changed = false;
        this.context = new import0.NgSwitchCase(p0, p1, p2);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_NgSwitchCase.prototype.check_ngSwitchCase = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.ngSwitchCase = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgSwitchCase.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_NgSwitchCase.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgSwitchCase;
}());
exports.Wrapper_NgSwitchCase = Wrapper_NgSwitchCase;
var Wrapper_NgSwitchDefault = (function () {
    function Wrapper_NgSwitchDefault(p0, p1, p2) {
        this.changed = false;
        this.context = new import0.NgSwitchDefault(p0, p1, p2);
    }
    Wrapper_NgSwitchDefault.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_NgSwitchDefault.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgSwitchDefault;
}());
exports.Wrapper_NgSwitchDefault = Wrapper_NgSwitchDefault;
