// @angular/common/src/directives/ng_style.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/common/src/directives/ng_style');
var import1 = require('@angular/core/src/change_detection/change_detection');
var import2 = require('@angular/core/src/linker/view_utils');
var Wrapper_NgStyle = (function () {
    function Wrapper_NgStyle(p0, p1, p2) {
        this.changed = false;
        this.context = new import0.NgStyle(p0, p1, p2);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_NgStyle.prototype.check_ngStyle = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.ngStyle = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgStyle.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        if (!throwOnChange) {
            this.context.ngDoCheck();
        }
        return changed;
    };
    Wrapper_NgStyle.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgStyle;
}());
exports.Wrapper_NgStyle = Wrapper_NgStyle;
