// @angular/http/src/http_module.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var import0 = require('@angular/core/src/linker/ng_module_factory');
var import1 = require('@angular/http/src/http_module');
var import2 = require('@angular/http/src/backends/browser_xhr');
var import3 = require('@angular/http/src/base_response_options');
var import4 = require('@angular/http/src/backends/xhr_backend');
var import5 = require('@angular/http/src/base_request_options');
var import7 = require('@angular/http/src/interfaces');
var import8 = require('@angular/http/src/http');
var import9 = require('@angular/http/src/backends/browser_jsonp');
var import10 = require('@angular/http/src/backends/jsonp_backend');
var HttpModuleInjector = (function (_super) {
    __extends(HttpModuleInjector, _super);
    function HttpModuleInjector(parent) {
        _super.call(this, parent, [], []);
    }
    Object.defineProperty(HttpModuleInjector.prototype, "_BrowserXhr_1", {
        get: function () {
            if ((this.__BrowserXhr_1 == null)) {
                (this.__BrowserXhr_1 = new import2.BrowserXhr());
            }
            return this.__BrowserXhr_1;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HttpModuleInjector.prototype, "_ResponseOptions_2", {
        get: function () {
            if ((this.__ResponseOptions_2 == null)) {
                (this.__ResponseOptions_2 = new import3.BaseResponseOptions());
            }
            return this.__ResponseOptions_2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HttpModuleInjector.prototype, "_XSRFStrategy_3", {
        get: function () {
            if ((this.__XSRFStrategy_3 == null)) {
                (this.__XSRFStrategy_3 = import1._createDefaultCookieXSRFStrategy());
            }
            return this.__XSRFStrategy_3;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HttpModuleInjector.prototype, "_XHRBackend_4", {
        get: function () {
            if ((this.__XHRBackend_4 == null)) {
                (this.__XHRBackend_4 = new import4.XHRBackend(this._BrowserXhr_1, this._ResponseOptions_2, this._XSRFStrategy_3));
            }
            return this.__XHRBackend_4;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HttpModuleInjector.prototype, "_RequestOptions_5", {
        get: function () {
            if ((this.__RequestOptions_5 == null)) {
                (this.__RequestOptions_5 = new import5.BaseRequestOptions());
            }
            return this.__RequestOptions_5;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HttpModuleInjector.prototype, "_Http_6", {
        get: function () {
            if ((this.__Http_6 == null)) {
                (this.__Http_6 = import1.httpFactory(this._XHRBackend_4, this._RequestOptions_5));
            }
            return this.__Http_6;
        },
        enumerable: true,
        configurable: true
    });
    HttpModuleInjector.prototype.createInternal = function () {
        this._HttpModule_0 = new import1.HttpModule();
        return this._HttpModule_0;
    };
    HttpModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import1.HttpModule)) {
            return this._HttpModule_0;
        }
        if ((token === import2.BrowserXhr)) {
            return this._BrowserXhr_1;
        }
        if ((token === import3.ResponseOptions)) {
            return this._ResponseOptions_2;
        }
        if ((token === import7.XSRFStrategy)) {
            return this._XSRFStrategy_3;
        }
        if ((token === import4.XHRBackend)) {
            return this._XHRBackend_4;
        }
        if ((token === import5.RequestOptions)) {
            return this._RequestOptions_5;
        }
        if ((token === import8.Http)) {
            return this._Http_6;
        }
        return notFoundResult;
    };
    HttpModuleInjector.prototype.destroyInternal = function () {
    };
    return HttpModuleInjector;
}(import0.NgModuleInjector));
exports.HttpModuleNgFactory = new import0.NgModuleFactory(HttpModuleInjector, import1.HttpModule);
var JsonpModuleInjector = (function (_super) {
    __extends(JsonpModuleInjector, _super);
    function JsonpModuleInjector(parent) {
        _super.call(this, parent, [], []);
    }
    Object.defineProperty(JsonpModuleInjector.prototype, "_BrowserJsonp_1", {
        get: function () {
            if ((this.__BrowserJsonp_1 == null)) {
                (this.__BrowserJsonp_1 = new import9.BrowserJsonp());
            }
            return this.__BrowserJsonp_1;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(JsonpModuleInjector.prototype, "_ResponseOptions_2", {
        get: function () {
            if ((this.__ResponseOptions_2 == null)) {
                (this.__ResponseOptions_2 = new import3.BaseResponseOptions());
            }
            return this.__ResponseOptions_2;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(JsonpModuleInjector.prototype, "_JSONPBackend_3", {
        get: function () {
            if ((this.__JSONPBackend_3 == null)) {
                (this.__JSONPBackend_3 = new import10.JSONPBackend_(this._BrowserJsonp_1, this._ResponseOptions_2));
            }
            return this.__JSONPBackend_3;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(JsonpModuleInjector.prototype, "_RequestOptions_4", {
        get: function () {
            if ((this.__RequestOptions_4 == null)) {
                (this.__RequestOptions_4 = new import5.BaseRequestOptions());
            }
            return this.__RequestOptions_4;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(JsonpModuleInjector.prototype, "_Jsonp_5", {
        get: function () {
            if ((this.__Jsonp_5 == null)) {
                (this.__Jsonp_5 = import1.jsonpFactory(this._JSONPBackend_3, this._RequestOptions_4));
            }
            return this.__Jsonp_5;
        },
        enumerable: true,
        configurable: true
    });
    JsonpModuleInjector.prototype.createInternal = function () {
        this._JsonpModule_0 = new import1.JsonpModule();
        return this._JsonpModule_0;
    };
    JsonpModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import1.JsonpModule)) {
            return this._JsonpModule_0;
        }
        if ((token === import9.BrowserJsonp)) {
            return this._BrowserJsonp_1;
        }
        if ((token === import3.ResponseOptions)) {
            return this._ResponseOptions_2;
        }
        if ((token === import10.JSONPBackend)) {
            return this._JSONPBackend_3;
        }
        if ((token === import5.RequestOptions)) {
            return this._RequestOptions_4;
        }
        if ((token === import8.Jsonp)) {
            return this._Jsonp_5;
        }
        return notFoundResult;
    };
    JsonpModuleInjector.prototype.destroyInternal = function () {
    };
    return JsonpModuleInjector;
}(import0.NgModuleInjector));
exports.JsonpModuleNgFactory = new import0.NgModuleFactory(JsonpModuleInjector, import1.JsonpModule);
